python/ql/test/query-tests/Security/CWE-327/pyOpenSSL_fluent.py

```diff
@@ -0,0 +1,41 @@
+import socket
+from OpenSSL import SSL
+
+def test_fluent():
+    hostname = 'www.python.org'
+    context = SSL.Context(SSL.SSLv23_METHOD)
+
+    conn = SSL.Connection(context, socket.socket(socket.AF_INET, socket.SOCK_STREAM))
+    r = conn.connect((hostname, 443))
+    print(conn.get_protocol_version_name())
+
+
+def test_fluent_no_TLSv1():
+    hostname = 'www.python.org'
+    context = SSL.Context(SSL.SSLv23_METHOD)
+    context.set_options(SSL.OP_NO_TLSv1)
+
+    conn = SSL.Connection(context, socket.socket(socket.AF_INET, socket.SOCK_STREAM))
+    r = conn.connect((hostname, 443))
+    print(conn.get_protocol_version_name())
+
+
+def test_fluent_safe():
+    hostname = 'www.python.org'
+    context = SSL.Context(SSL.SSLv23_METHOD)
+    context.set_options(SSL.OP_NO_SSLv2)
+    context.set_options(SSL.OP_NO_SSLv3)
+    context.set_options(SSL.OP_NO_TLSv1)
+    context.set_options(SSL.OP_NO_TLSv1_1)
```

Comment:
I was confused when looking this over, since this doesn't match what the `ssl` module form standard library does. and was about to suggest that we could remove the `OP_NO_SSLv2` and `OP_NO_SSLv3`... but then I realized that I requested that these be added myself in https://github.com/github/codeql/pull/5284#discussion_r597520694

Currently I have no clue how I figured out that the old part (that only disabled TLSv1 and TLSv1.1) was not enough. However, after manually installing `pyOpenSSL` PyPI package, I can confirm that `SSL.OP_NO_SSLv3` is not set by default (on my system with OpenSSL 1.1.1):

```py
from OpenSSL import SSL
c = SSL.Context(SSL.SSLv23_METHOD)
current_options_bitmask = SSL._lib.SSL_CTX_get_options(c._context)
print(current_options_bitmask, SSL.OP_NO_SSLv3, current_options_bitmask & SSL.OP_NO_SSLv3) # last value printed is 0, indicating that the option to disable SSLv3 is not enabled. Not that on my system, `SSL.OP_NO_SSLv2 == 0`, since my version of OpenSSL does not support SSLv2 at all.
```

Reply:
Thanks for checking!